framework/test/gtest/BundleHooksTest.cpp

```diff
@@ -161,24 +161,78 @@ TEST_F(BundleHooksTest, TestFindHook)
 
     // Test for non-filtered GetBundle(long) result
     ASSERT_TRUE(framework.GetBundleContext().GetBundle(bundleAId));
+    ASSERT_TRUE(bundleA.GetBundleContext().GetBundle(bundleAId));
 
     auto findHookReg
         = framework.GetBundleContext().RegisterService<BundleFindHook>(std::make_shared<TestBundleFindHook>());
 
     // Test for filtered GetBundle(long) result
-    ASSERT_FALSE(framework.GetBundleContext().GetBundle(bundleAId));
+    ASSERT_TRUE(framework.GetBundleContext().GetBundle(bundleAId)); // framework context should NEVER filter
+    ASSERT_FALSE(bundleA.GetBundleContext().GetBundle(bundleAId));
 
     auto bundles = framework.GetBundleContext().GetBundles();
+    bool foundBundle = false;
     for (auto const& i : bundles)
     {
-        ASSERT_NE(i.GetSymbolicName(), "TestBundleA");
+        if(i.GetSymbolicName() == "TestBundleA"){
+            foundBundle = true;
+        }
     }
+    ASSERT_TRUE(foundBundle);
+
+    bundles = bundleA.GetBundleContext().GetBundles();
+    foundBundle = false;
+    for (auto const& i : bundles)
+    {
+        if(i.GetSymbolicName() == "TestBundleA"){
+            foundBundle = true;
+        }
+    }
+    ASSERT_FALSE(foundBundle);
+
+    findHookReg.Unregister();
+
+    bundleA.Stop();
+}
+
+TEST_F(BundleHooksTest, TestFindHookBundleInstall)
+{
+    // install to get diff bundleContext than framework
+    auto bundleB = cppmicroservices::testing::InstallLib(framework.GetBundleContext(), "TestBundleB");
+    ASSERT_TRUE(bundleB);
+    bundleB.Start();
+
+    auto findHookReg
+        = framework.GetBundleContext().RegisterService<BundleFindHook>(std::make_shared<TestBundleFindHook>());
+    Bundle bundleA;
+    
+    // if NOT building shared_libs
+        // install just invokes getBundles which will filter always
+        // because it invoked getBundles, nothing is thrown... that only happens on an actual instal
+#if defined(US_BUILD_SHARED_LIBS)
+    // on first installation, the bundle will not exist and therefore will be installed regardless of hooks
+    bundleA = cppmicroservices::testing::InstallLib(bundleB.GetBundleContext(), "TestBundleA");
+    ASSERT_TRUE(bundleA);
+    ASSERT_EQ(bundleA.GetSymbolicName(), "TestBundleA");
+
+
+    // now that it exists, if installed with non-system bundle it will fail
+    ASSERT_THROW(cppmicroservices::testing::InstallLib(bundleB.GetBundleContext(), "TestBundleA"), std::runtime_error);
```

Comment:
Given that the `TestBundleFindHook` is being used and it doesn't throw, I'm not sure I see how this line throws an exception. Shouldn't this line not throw an exception?